### PR TITLE
helper: move config_for_role_exists from horizon to crowbar-openstack (SOC-10191)

### PR DESCRIPTION
**Pull request by gosipyan**
This is clean up ,as we need to use helper outside horizon barclamp
would be more convenient  to move it to crowbar-openstack.

### CODE REVIEW

**Comment by sjamgade** [comment]
the PR is failing CI because 49ab94a821 does not exist in to stable/5.0-pike branch

File: chef/cookbooks/crowbar-openstack/libraries/helpers.rb [hunk 1]
```diff
@@ -300,6 +304,13 @@ def self.get_nodes(node, role, barclamp, instance)
     nodes
   end
 
+  def self.config_for_role_exists?(name)
+    shouldbe = "#{name}-config-"
+    @cached_roles ||= Chef::Role.list.keys
+    res = @cached_roles.find { |rname| rname.start_with? shouldbe }
+    res != nil
+  end
```
Comment (rsalevsky) [inline chef/cookbooks/crowbar-openstack/libraries/helpers.rb]: empty line missing below

Comment (gosipyan) [reply chef/cookbooks/crowbar-openstack/libraries/helpers.rb]: done

**Comment by dirkmueller** [comment]
@gosipyan it looks like the CI failure is related:

```
 ================================================================================
15:17:16 Recipe Compile Error in /var/chef/cache/cookbooks/horizon/recipes/role_horizon_server.rb
15:17:16 ================================================================================
15:17:16 
15:17:16 NoMethodError
15:17:16 -------------
15:17:16 private method `config_for_role_exists?' called for CrowbarOpenStackHelper:Class
15:17:16 
15:17:16 Cookbook Trace:
15:17:16 ---------------
15:17:16   /var/chef/cache/cookbooks/crowbar-openstack/libraries/helpers.rb:42:in `config_for_role_exists?'
15:17:16   /var/chef/cache/cookbooks/horizon/recipes/server.rb:598:in `from_file'
15:17:16   /var/chef/cache/cookbooks/horizon/recipes/role_horizon_server.rb:18:in `from_file'
15:17:16 
15:17:16 Relevant File Content:

```

**Pull request merged**